src/send/SendEnterAmount.tsx

```diff
@@ -37,38 +43,59 @@ import { NETWORK_NAMES } from 'src/shared/conts'
 import Colors from 'src/styles/colors'
 import { typeScale } from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
-import { useTokenInfo, useTokenToLocalAmount } from 'src/tokens/hooks'
+import { useTokenToLocalAmount } from 'src/tokens/hooks'
 import { tokensWithNonZeroBalanceAndShowZeroBalanceSelector } from 'src/tokens/selectors'
 import { TokenBalance } from 'src/tokens/slice'
 import { getSupportedNetworkIdsForSend } from 'src/tokens/utils'
+import { isDekRegisteredSelector, walletAddressSelector } from 'src/web3/selectors'
+import { usePrepareSendTransactions } from 'src/send/usePrepareSendTransactions'
+import Logger from 'src/utils/Logger'
 
 type Props = NativeStackScreenProps<StackParamList, Screens.SendEnterAmount>
 
 const TOKEN_SELECTOR_BORDER_RADIUS = 100
 const MAX_BORDER_RADIUS = 96
 
-// This is just a mock implementation to test various error states
-// TODO(ACT-955): replace with real implementation
-function usePrepareTransactions(amount: BigNumber, token: TokenBalance) {
-  const nativeToken = useTokenInfo(`${token.networkId}:native`)
-  if (!amount || amount.eq(0)) {
-    return
-  }
-  if (token.isNative && token.balance.minus(amount).lt(0.1)) {
-    return {
-      type: 'need-decrease-spend-amount-for-gas' as const,
-      feeCurrency: token,
-    }
-  }
-  if (!token.isNative && nativeToken?.balance.lt(0.1)) {
-    return {
-      type: 'not-enough-balance-for-gas' as const,
-      feeCurrencies: [nativeToken],
-    }
-  }
-  return {
-    type: 'possible' as const,
-  }
+const TAG = 'SendEnterAmount'
+
+const FETCH_UPDATED_TRANSACTIONS_DEBOUNCE_TIME = 250
+
+function FeeLoading() {
+  return (
+    <>
+      <View testID="SendEnterAmount/FeeLoading" style={styles.feeInCryptoContainer}>
+        <Text style={styles.feeInCrypto}>{'≈ '}</Text>
+        <ActivityIndicator size="small" color={Colors.gray4} />
+      </View>
+    </>
+  )
+}
+
+function FeePlaceholder({ feeTokenSymbol }: { feeTokenSymbol: string }) {
+  return (
+    <>
+      <Text testID="SendEnterAmount/FeePlaceholder" style={styles.feeInCrypto}>
+        ~ {feeTokenSymbol}
+      </Text>
+    </>
+  )
+}
+
+function FeeAmount({ feeTokenId, feeAmount }: { feeTokenId: string; feeAmount: BigNumber }) {
+  return (
+    <>
+      <View testID="SendEnterAmount/FeeInCrypto" style={styles.feeInCryptoContainer}>
+        <Text style={styles.feeInCrypto}>~</Text>
```

Comment:
Finn added a `showApprox` prop to TokenDisplay on the send confirmation screen updates PR, so we can drop this Text and the wrapper view and just use that.